docs/deprecated-features/index.md

```diff
@@ -0,0 +1,99 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Deprecated Features
+
+## Overview
+
+Over time, some features of RabbitMQ are not relevant anymore. Sometimes, a
+feature must go away to leave the room for a better replacement. Before we
+deorbit a feature, we follow a deprecation time window: we announce that a
+feature is deprecated along with the reasoning, how user should prepare for
+that and the estimated future version of RabbitMQ where this deprecated
+feature will be removed and unsupported.
+
+Beside the public announcement, RabbitMQ has a Deprecated features subsystem
+that helps notify users right from the broker if they are using a feature that
+will be removed in the future.
+
+## The lifecycle of a deprecated feature
+
+The Deprecated features subsystem is based on the [Feature flags
+subsystem](./feature-flags). Therefore, much like a feature flag which goes
+from experimental to stable to required, a deprecated feature follow the
+following phases:
+
+1. **Permitted by default**. A user can continue to use the feature. They just
+   get warnings in the logs and can use a configuration knob to start RabbitMQ
+   as if the feature was removed for testing purpose.
+2. **Denied by default**. The feature is still there, but it is not usable out
+   of the box, like if it was already removed. A user can turn it back on from
+   the configuration.
+3. **Disconnected**. The feature is still present in the code, but it is
+   compiled out in official packages. A user has to recompile RabbitMQ from
+   source to be able to continue to use it. Note that we do not plan to use
+   this phase very often, if ever.
+4. **Removed**. The code of the feature is gone and it is not possible to use
+   it anymore. This is the equivalent of a required feature flag.
+
+## Checking if deprecated features are used
+
+The CLI provides a command that lists if some deprecated features are used by
+a running node or cluster:
+
+```sh
+rabbitmq-diagnostics check_if_any_deprecated_features_are_used
+```
+
+It lists used deprecated features (if any) and exits with 0 if none are used,
+or non-zero if one or more deprecated featurs are actively used.
+
+There is also an HTTP API endpoing to list used deprecated features:
+
+```
+GET /api/deprecated-features/used
+```
+
+:::info
+Note that the result of these tools depends on if the deprecated feature use
+can be detected or not. For instance, it was possible to detect if classic
+queue mirroring was used. However it is not possible for global QoS.
+:::
+
+## Configuring deprecated features behavior
+
+As described above, a user may decide to turn off a deprecated feature to test
+the behavior of RabbitMQ and their applications as if the feature was already
+removed.
+
+```ini
+# Try RabbitMQ as if the deprecated feature was gone.
+deprecated_features.permit.some_deprecated_feature = false
```

Comment:
We should list the names of such features in 3.13.x and 4.0.x.

Reply:
Ok, adding the list. Thank!

Reply:
The list is now available in a dedicated page at `/docs/next/deprecated-features/list`.